test/algorithm/copy_backward.cpp

```diff
@@ -40,7 +40,7 @@ int main()
     CHECK(std::equal(a, a + size(a), out));
 
     std::fill_n(out, size(out), std::make_pair(0, 0));
-    auto res2 = ranges::copy_backward(ranges::view::all(a), end(out));
+    auto res2 = ranges::copy_backward(ranges::make_iterator_range(a), end(out));
```

Comment:
Why?

Reply:
These algorithm tests ensure that the algorithm returns a `dangling` result when passed an argument that is not a `ReferenceableRange`. `ref_view` and `subrange` both satisfy `ReferenceableRange`, so using `make_subrange` here would break the test just the same as does using `view:all`. Since `iterator_range` doesn't satisfy `ReferenceableRange`, and we're keeping it around for backwards compatibility, this was an easy fix. (Although I note in this specific instance that `a` is a container so `std::move(a)` would work as well.)

In cmcstl2 I fixed the algorithm tests by adding a test-only `unref_view` view adaptor that simply wraps the view with which it's initialized. I could/should do the same here, which would (a) be consistent across both libraries, which is valuable in itself, and (b) allow us to deprecate `iterator_range` with `RANGES_DEPRECATE` if you'd like to push people to `subrange` more quickly.

Reply:
Yeah... about that.... https://github.com/CaseyCarter/cmcstl2/pull/126#pullrequestreview-85859245